### PR TITLE
Tgui light mode

### PR DESCRIPTION
**Pull request by pali6**
## About the PR 
Makes it so there's a framework for adding light mode variants of tgui themes. Also adds a light mode variant for the default (nanotrasen) and the NTOS (used in chem dispenser) themes. Light modes for other themes can be relatively easily added later.

(I don't think this is a screenshot of the latest version but it's close enough.)
dreamseeker_lzGu73wR1O

Technically this also adds supports for more modes than just light and dark.

## Why's this needed? 
Consistency is good and some people use light mode either due to preference or due to accessibility issues so why not have light mode UIs too.

## Changelog 

```changelog
(u)pali
(*)Light mode users will now have light mode applied to (most) TGUI windows too
```

### CODE REVIEW

File: code/modules/tgui/tgui.dm [hunk 1]
```diff
@@ -206,6 +206,7 @@
 			"size" = window_size,
 			"fancy" = user.client.preferences.tgui_fancy,
 			"locked" = user.client.preferences.tgui_lock,
+			"mode" = user.client.darkmode ? "dark" : "light",
```
Comment (ZeWaka) [inline code/modules/tgui/tgui.dm:209]:
```suggestion
			"mode" = user.client.darkmode ? "dark" : "light", // |GOONSTATION-ADD|
```

File: tgui/packages/tgui/layouts/Layout.js [hunk 2]
```diff
@@ -12,11 +12,12 @@ export const Layout = props => {
   const {
     className,
     theme = 'nanotrasen',
+    mode = 'dark',
```
Comment (ZeWaka) [inline tgui/packages/tgui/layouts/Layout.js:15]:
```suggestion
    mode = 'dark', /* |GOONSTATION-ADD| */
```

File: tgui/packages/tgui/layouts/Window.js [hunk 3]
```diff
@@ -74,6 +74,7 @@ export class Window extends Component {
     const { debugLayout } = useDebug(this.context);
     const dispatch = useDispatch(this.context);
     const fancy = config.window?.fancy;
+    const mode = config.window?.mode;
```
Comment (ZeWaka) [inline tgui/packages/tgui/layouts/Window.js:77]:
```suggestion
    const mode = config.window?.mode; /* |GOONSTATION-ADD| */
```

**Review by ZeWaka — COMMENTED** [review]
should be valid code

**Review by mordent-goonstation — APPROVED** [review]
Sure, why not? Broadly trusting the colors are fine, I'm not going to manually look through them.

Only possible thought is whether it makes sense to manually style the NanoTrasen logo with a fill color rather than have a whole separate logo. Not too opposed to having a distinct logo.

**Comment by pali6** [comment]
> manually style the NanoTrasen logo with a fill color

How would I go about that?

**Comment by mordent-goonstation** [comment]
> > manually style the NanoTrasen logo with a fill color
> 
> How would I go about that?

Looking at the implementation, it's done as a `background-image` which complicates things. You could get away with using the image as a mask over a `background-color` or something, but probably overcomplicated for the sake of "oh no, two logo files!".

So good as is.

**Pull request merged**